portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +26,34 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    public List<String> messages;
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ciara!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html;");
+        response.getWriter().println("Hello Ciara!");
+
+        String json = convertToJsonUsingGson(messages);
+
+        response.setContentType("application/json;");
+        response.getWriter().println(messages);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = request.getParameter("box");
```

Comment:
"box" and "comment" etc are too generic. In this situation it is okay. However, in real world, please use more descriptive names to enhance readability.